### PR TITLE
feat(72): Improved convenience of sdrun in interactive mode 

### PR DESCRIPTION
**Pull request by yakanechi**
## Context

Extend the sdrun command in interactive mode (added with #95) 
This command did not have a list of steps, tabbed completion, etc.

## Objective

- Add step completion by Tab key
- Add `--help`,`--list` and `--all` flags
  - `--all` executes all steps in order of definition. And if it fails in the middle of the process, it will stop there.

## References

#72 
#95

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: launch/docker.go [hunk 1]
```diff
@@ -282,7 +334,8 @@ func (d *docker) runBuild(buildEntry buildEntry) error {
 			{".", "/tmp/sd-local.env"},
 			{"set", "+a"},
 			{"export", "PS1='sd-local# '"},
-			{"sdrun() { . /$SD_UTILS_DIR/bin/sd-run $@; }"},
+			{"sdrun() { . /$SD_UTILS_DIR/bin/sdrun $@; }"},
+			{"source /$SD_UTILS_DIR/bin/sdrun-bash-completion"},
```
Comment (y-oksaku) [inline launch/docker.go:338]: Probably, `source`  command is not available in some docker image with using `/bin/sh`.
You need to use `.` (dot) command.

File: launch/docker.go [hunk 2]
```diff
@@ -138,20 +138,72 @@ func (d *docker) setupInteractiveMode(buildEntry *buildEntry) error {
...
+		`    echo "Usage:"`,
+		`    echo "  sdrun [flags]"`,
+		`    echo "  sdrun [step]"`,
+		`    echo ""`,
+		`    echo "Flags:"`,
+		`    echo "  --help     help for sdrun"`,
+		`    echo "  --list     show all steps"`,
+		`    echo "  --all      run all steps"`,
+		`elif [ "$1" = "--list" ]; then`,
+		`  echo "${step_list}";`,
```
Comment (y-oksaku) [inline launch/docker.go:158]: It would be better to show step names in order of the definition of `screwdriver.yaml`.

File: launch/docker.go [hunk 3]
```diff
@@ -138,20 +138,72 @@ func (d *docker) setupInteractiveMode(buildEntry *buildEntry) error {
...
+		`elif [ "$1" = "--all" ]; then`,
+		`  steps=$(cat "${step_dir}/.steps")`,
+		`  for step_name in $steps; do`,
+		`    cat "${step_dir}/${step_name}" | sed "s/^/${step_name}: /"`,
+		fmt.Sprintf(`    %s "${step_dir}/${step_name}" | sed "s/^/${step_name}: /"`, shellBin),
```
Comment (y-oksaku) [inline launch/docker.go:163]: Using `sh <step-file>` makes scripts closed.
So you cannot use the variable defined in step script file in the other steps.

File: launch/docker.go (hunk 1 above)
Comment (yakanechi) [reply launch/docker.go]: fixed

File: launch/docker.go (hunk 2 above)
Comment (yakanechi) [reply launch/docker.go]: fixed

File: launch/docker.go (hunk 3 above)
Comment (yakanechi) [reply launch/docker.go]: fixed

**Review by y-oksaku — APPROVED** [review]

**Pull request merged**